crates/sui-adapter-transactional-tests/tests/programmable_transaction_examples/publish.move

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// simple test of publish
+
+//# init --addresses p=0x0 q=0x0 r=0x0 --accounts A
+
+//# package
+module p::m {
+    public fun foo(x: u64) {
+        p::n::bar(x)
```

Comment:
Outside of the test setting, we require that modules are supplied in reverse dependency order, which would imply `p::n` before `p::m` here -- the `Publish` command should fail to verify `p::m` otherwise.  Are we doing something special during compilation to avoid that restriction?

Reply:
Yes, we output the modules in topological order (or reverse topological order (or whatever it is you need))